lib/cpp/src/thrift/transport/TBufferTransports.h

```diff
@@ -587,11 +587,9 @@ class TMemoryBuffer : public TVirtualTransport<TMemoryBuffer, TBufferBase> {
   void resetBuffer() {
     rBase_ = buffer_;
     rBound_ = buffer_;
-    wBase_ = buffer_;
-    // It isn't safe to write into a buffer we don't own.
-    if (!owner_) {
-      wBound_ = wBase_;
-      bufferSize_ = 0;
+    if (owner_) {
+      wBase_ = buffer_;
+      wBound_ = buffer_ + bufferSize_;
```

Comment:
Could you explain this change a bit? Originally the code always set `wBase_ = buffer_;` but after your change, `wBase_` is only set if the buffer is owned. Is that guaranteed correct?

Reply:
If the buffer is not owned (read-only), wbase is initialized to buffer + wpos (the end of the buffer) and never updated, along with wbound. Throughout the lifetime of the buffer, if it is not owned (read-only), wbase == wbound meaning there is no space to write. Only if the buffer is owned (writeable) should wbase and wbound be reset to the beginning / end of the buffer.

Reply:
Please let me know if you would like additional testing done to illustrate things a bit more.